pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -363,7 +364,10 @@ func (p *criStatsProvider) getFsInfo(fsID *runtimeapi.FilesystemIdentifier) *cad
 	fsInfo, err := p.cadvisor.GetDirFsInfo(mountpoint)
 	if err != nil {
 		msg := "Failed to get the info of the filesystem with mountpoint"
-		if err == cadvisorfs.ErrNoSuchDevice {
+		noDataFound := errors.Is(err, memory.ErrDataNotFound) ||
+			// from: https://github.com/kubernetes/kubernetes/blob/cf3374e/vendor/github.com/google/cadvisor/fs/fs.go#L593
+			strings.HasSuffix(err.Error(), "in cached partitions map")
```

Comment:
Does it make sense to return a custom error from cadvisor for this?

Reply:
I think so. I proposed the change at https://github.com/google/cadvisor/pull/2859

We may wanna switch as follow-up to that type to unblock this PR?

Reply:
That has merged, can we update?

Reply:
Yes, changed to validate the error from cadvisor.

Reply:
Ugh, we have to move the error into a place without build constraints. Following-up in https://github.com/google/cadvisor/pull/3018